lib/HLSL/HLOperations.cpp

```diff
@@ -289,6 +292,17 @@ void SetHLLowerStrategy(Function *F, StringRef S) {
   F->addFnAttr(HLLowerStrategy, S);
 }
 
+// Set function attribute indicating wave-sensitivity
+void SetHLWaveSensitive(Function *F) {
+  F->addFnAttr(HLWaveSensitive, "y");
```

Comment:
The only checks made on this attribute are whether it's present. The "y" doesn't matter, but it's informative.